pkg/controller/stackconfigpolicy/controller.go

```diff
@@ -0,0 +1,465 @@
+		MatchLabels:      policy.Spec.ResourceSelector.MatchLabels,
+		MatchExpressions: policy.Spec.ResourceSelector.MatchExpressions,
+	})
+	if err != nil {
+		return results.WithError(err), status
+	}
+	listOpts := client.ListOptions{LabelSelector: selector}
+
+	// restrict the search to the policy namespace if it is different from the operator namespace
+	if policy.Namespace != r.params.OperatorNamespace {
+		listOpts.Namespace = policy.Namespace
+	}
+
+	// find the list of Elasticsearch to configure
+	var esList esv1.ElasticsearchList
+	if err := r.Client.List(ctx, &esList, &listOpts); err != nil {
+		return results.WithError(err), status
+	}
+
+	configuredResources := esMap{}
+	for _, es := range esList.Items {
+		log.V(1).Info("Reconcile StackConfigPolicy", "policy_namespace", policy.Namespace, "policy_name", policy.Name, "es_namespace", es.Namespace, "es_name", es.Name)
+		es := es
+
+		// keep the list of ES to be configured
+		esNsn := k8s.ExtractNamespacedName(&es)
+		configuredResources[esNsn] = es
+
+		// File based Settings is available from ES 8.5.0
+		v, err := version.Parse(es.Spec.Version)
+		if err != nil {
+			return results.WithError(err), status
+		}
+		if v.LT(fileBasedSettingsMinimumVersion) {
+			err = fmt.Errorf("invalid version to configure resource Elasticsearch %s/%s: actual %s, expected >= %s", es.Namespace, es.Name, v, fileBasedSettingsMinimumVersion)
+			r.recorder.Eventf(&policy, corev1.EventTypeWarning, events.EventReasonUnexpected, err.Error())
+			results.WithError(err)
+			status.AddPolicyErrorFor(esNsn, policyv1alpha1.ErrorPhase, err.Error())
+			continue
+		}
+
+		// the file Settings Secret must exist, if not it will be created empty by the ES controller
+		var secret corev1.Secret
+		err = r.Client.Get(ctx, types.NamespacedName{Namespace: es.Namespace, Name: esv1.FileSettingsSecretName(es.Name)}, &secret)
+		if err != nil && apierrors.IsNotFound(err) {
+			// requeue if the Secret has not been created yet
+			return results.WithResult(defaultRequeue), status
+		}
+		if err != nil {
+			return results.WithError(err), status
+		}
+
+		current, err := filesettings.NewSettingsSecretFromSecret(secret)
+		if err != nil {
+			return results.WithError(err), status
+		}
+
+		// check that there is no other policy that already owns the Settings Secret
+		currentOwner, ok := current.CanBeOwnedBy(policy)
+		if !ok {
+			err = fmt.Errorf("conflict: resource Elasticsearch %s/%s already configured by StackConfigpolicy %s/%s", es.Namespace, es.Name, currentOwner.Namespace, currentOwner.Name)
+			r.recorder.Eventf(&policy, corev1.EventTypeWarning, events.EventReasonUnexpected, err.Error())
+			results.WithError(err)
+			status.AddPolicyErrorFor(esNsn, policyv1alpha1.ConflictPhase, err.Error())
+			continue
+		}
+
+		// create the expected Settings Secret
+		newVersion := time.Now().UnixNano()
+		expected, err := filesettings.NewSettingsSecret(newVersion, &current, esNsn, &policy)
```

Comment:
I made the version external to `NewSettingsSecret` to make it easier to test, following https://github.com/elastic/cloud-on-k8s/pull/6148#discussion_r1017747081 in 7f9af81ae1cd.

Reply:
I thought that relying on a non-monotonic clock in a test could introduce (very theoretical) instability and it would be cleaner to factor it out.

Reply:
Another idea if we want to avoid the external creation of the argument is to have a secondary constructor `NewSettingsSecretWithVersion` or something like that that calls the existing constructor function which is also the one we test.